python/lsst/ts/electrometer/controller.py

```diff
@@ -45,158 +51,159 @@ def __init__(self):
+        self.commander.write(f"{self.commands.perform_zero_calibration(self.mode, self.auto_range, self.range)}\r".encode())
+
+    def set_digital_filter(self, activate_filter, activate_avg_filter, activate_med_filter):
+        filter = activate_filter
+        if activate_avg_filter is True and activate_filter is False:
+            filter = False
+        if activate_avg_filter is False and activate_filter is True:
+            filter = False
+        self.commander.write(f"{self.commands.activate_filter(self.mode, Filter(2), filter)}\r".encode())
+        filter = activate_filter
+        if activate_med_filter is True and activate_filter is False:
+            filter = False
+        if activate_med_filter is False and activate_filter is True:
+            filter = False
+        self.commander.write(f"{self.commands.activate_filter(self.mode, Filter(1), filter)}\r".encode())
+        self.check_error()
+
+    def set_integration_time(self, int_time):
+        self.commander.write(f"{self.commands.integration_time(mode=self.mode, time=int_time)}\r".encode())
+        self.check_error()
 
     def set_mode(self, mode):
-        self.mode = UnitMode(mode)
+        self.commander.write(f"{self.commands.set_mode(mode=mode)}\r".encode())
+        self.check_error()
 
     def set_range(self, set_range):
-        self.range = set_range
-
-    async def start_scan(self):
-        while not self.stop_reading_value:
-            self.buffer.put(self.value(intensity=0, time=0, temperature=0, unit=self.mode.name))
-            print(f"{self.buffer}")
-            await asyncio.sleep(self.integration_time)
-
-    async def start_scan_dt(self, scan_duration):
-        start = time.time()
+        self.commander.write(f"{self.commands.set_range(auto=self.auto_range, range_value=set_range, mode=self.mode)}\r".encode())
+        self.check_error()
+
+    def start_scan(self):
+        self.commander.write(f"{self.commands.clear_buffer()}\r".encode())
+        self.commander.write(f"{self.commands.format_trac()}\r".encode())
+        self.commander.write(f"{self.commands.set_buffer_size(50000)}\r".encode())
+        self.commander.write(f"{self.commands.select_device_timer()}\r".encode())
+        self.commander.write(f"{self.commands.next_read()}\r".encode())
+        self.manual_start_time = time.time()
+
+    def start_scan_dt(self, scan_duration):
+        self.commander.write(f"{self.commands.prepare_device_scan()}\r".encode())
+        self.manual_start_time = time.time()
         dt = 0
-        while not self.stop_reading_value or dt < scan_duration:
-            dt = time.time() - start
-            self.buffer.put(self.value(intensity=0, time=0, temperature=0, unit=self.mode.name))
-            await asyncio.sleep(self.integration_time)
+        # FIXME blocking and needs to be non-blocking
+        while dt < scan_duration:
+            time.sleep(self.integration_time)
+            dt = time.time() - self.manual_start_time
```

Comment:
Please use `time.monotonic()` instead of `time.time()` for elapsed time measurement.